drv/stm32h7-sprot-server/src/main.rs

```diff
@@ -0,0 +1,593 @@
+                // None of responses is expected to exceed a 2-byte payload
+                // plus trailing 2-byte checksum.
+                // The header and the first four bytes of the payload will be
+                // updated each time through the loop.
+                self.txmsg.init(MsgType::SinkReq, size);    // Tx msg minor update each iteration.
+                let mut n: u8 = self.txmsg.header_buf().len() as u8;
+                if let Some(buf) = self.txmsg.payload_buf_mut().get_mut(0..size) {
+                    buf.fill_with(|| {
+                        let seq = n;
+                        n = n.wrapping_add(1);
+                        seq
+                    });
+                }
+                let mut sent = 0u16;
+                const MAX_SINKREQ_ATTEMPTS: u16 = 4;
+                loop {
+                    if sent == count {
+                        break;
+                    }
+                    // For debugging: first two payload bytes are a message
+                    // sequence number if there is space for it.
+                    if let Some(seqno) = self.txmsg.payload_buf_mut().get_mut(0..2) {
+                        LittleEndian::write_u16(seqno, sent);
+                    }
+                    self.txmsg.set_crc();
+
+                    let result: Result<(), MsgError> = match self.do_send_recv_retries(TIMEOUT_QUICK, MAX_SINKREQ_ATTEMPTS) {
+                        Err(err) => Err(err),
+                        Ok(payload_len) => {
+                            match self.rxmsg.msgtype() {
+                                MsgType::SinkRsp => {
+                                    // TODO: Check sequence number in response.
+                                    Ok(())
+                                },
+                                MsgType::ErrorRsp if payload_len > 0 => {
+                                    if_chain! {
+                                        if let Some(bytes) = self.rxmsg.payload_buf();
+                                        if let Some(bytes) = bytes.get(0..1);
+                                        if let Some(code) = MsgError::from_u8(bytes[0]);
+                                        then {
+                                            Err(code)
+                                        } else {
+                                            Err(MsgError::Unknown)
+                                        }
+                                    }
+                                },
+                                MsgType::ErrorRsp => {
+                                    Err(MsgError::Unknown)
+                                },
+                                _ => {
+                                    // Other codes from RoT
+                                    // are not recoverable
+                                    // with a retry.
+                                    Err(MsgError::Unknown)
```

Comment:
Should this be 

```suggestion
                                    Err(MsgError::BadMessageType)
```
?

Reply:
Done